modules/vaos/app/services/eps/provider_service.rb

```diff
@@ -141,82 +154,57 @@ def address_matches?(provider, address)
 
       provider_address = provider[:location][:address]
 
-      # Step 1: Primary filter - check if street address matches (most likely to filter out non-matches)
-      return false unless street_address_matches?(provider_address, address[:street1])
-
-      # Step 2: Full address comparison with logging for monitoring, so we can detect formatting differences
-      full_match = full_address_matches?(provider_address, address)
+      # Compare the two reliable components: street and 5-digit zip
+      street_matches = street_address_matches?(provider_address, address[:street1])
+      zip_matches = zip_code_matches?(provider_address, address[:zip])
 
-      log_partial_address_match(provider_address, address) unless full_match
+      # Log for monitoring if some components match but not all (helps identify format issues)
+      if zip_matches && !street_matches
+        Rails.logger.warn("Provider address partial match - Street: #{street_matches}, " \
+                          "Zip: #{zip_matches}. " \
+                          "Provider: '#{provider_address}', " \
+                          "Referral: '#{address[:street1]}, #{address[:zip]}'")
+      end
 
-      # Return false if full address doesn't match
-      full_match
+      street_matches && zip_matches
     end
 
     ##
-    # Check if the street portion of provider address matches referral street1
+    # Check if street address matches by comparing referral street to beginning of provider address
     #
     # @param provider_address [String] Full provider address string
-    # @param referral_street1 [String] Street address from referral
-    # @return [Boolean] True if street addresses match
+    # @param referral_street [String] Street address from referral
+    # @return [Boolean] True if provider address starts with referral street
     #
-    def street_address_matches?(provider_address, referral_street1)
-      return false if provider_address.blank? || referral_street1.blank?
+    def street_address_matches?(provider_address, referral_street)
+      return false if provider_address.blank? || referral_street.blank?
 
-      # Check if referral street appears at the beginning of provider address
-      normalized_provider_address = normalize_address_text(provider_address)
-      normalized_referral_street = normalize_address_text(referral_street1)
+      normalized_provider = normalize_address_text(provider_address)
+      normalized_referral = normalize_address_text(referral_street)
```

Comment:
Note: We don't extract address parts as even the street address could have commas, so we just test the start of the address string.